pkg/controller/authenticationservice/authenticationservice_controller.go

```diff
@@ -26,15 +27,27 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+var (
+	authinfo = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "authentication_service_ready",
+			Help: "EnMasse authentication services in the ready state",
+		},
+		[]string{"authservice_name", "authservice_type"},
+	)
+)
+
 var log = logf.Log.WithName("controller_authenticationservice")
 
 // Gets called by parent "init", adding as to the manager
 func Add(mgr manager.Manager) error {
+	metrics.Registry.MustRegister(authinfo)
```

Comment:
Maybe create this in the reconciler so that its not reporting this metrics unless the reconciler was successfully created?